samples/nrf9160/modem_shell/src/drivers/nrf91_non_offloaded_dev.c

```diff
@@ -0,0 +1,507 @@
+	 * PDN type "IPV4V6": <IPv4>,<IPv6>, or <IPV4> or <IPv6>
+	 */
+	ret = nrf_modem_at_scanf(cmd, "+CGPADDR: %*d,\"%46[.:0-9A-F]\",\"%46[:0-9A-F]\"", addr1,
+				 addr2);
+	if (ret <= 0) {
+		return;
+	}
+
+	/* parse 1st IP string */
+	if (addr4 != NULL && inet_pton(AF_INET, addr1, tmp) == 1) {
+		strcpy(addr4, addr1);
+	} else if (addr6 != NULL && inet_pton(AF_INET6, addr1, tmp) == 1) {
+		strcpy(addr6, addr1);
+		return;
+	}
+
+	/* parse second IP string (IPv6 only) */
+	if (addr6 == NULL) {
+		return;
+	}
+	if (ret > 1 && inet_pton(AF_INET6, addr2, tmp) == 1) {
+		strcpy(addr6, addr2);
+	}
+}
+
+/******************************************************************************/
+
+struct pdn_conn_dyn_params {
+	struct in_addr dns_addr4_primary;
+	struct in_addr dns_addr4_secondary;
+	struct in6_addr dns_addr6_primary;
+	struct in6_addr dns_addr6_secondary;
+	uint32_t ipv4_mtu;
+	uint32_t ipv6_mtu;
+};
+#define AT_CMD_PDP_CONTEXT_READ_DYN_INFO "AT+CGCONTRDP=%d"
+#define AT_CMD_PDP_CONTEXT_READ_DYN_INFO_PARAM_COUNT 20
+#define AT_CMD_PDP_CONTEXT_READ_DYN_INFO_DNS_ADDR_PRIMARY_INDEX 6
+#define AT_CMD_PDP_CONTEXT_READ_DYN_INFO_DNS_ADDR_SECONDARY_INDEX 7
+#define AT_CMD_PDP_CONTEXT_READ_DYN_INFO_MTU_INDEX 12
+
+static int util_get_pdn_conn_dyn_params(int cid, struct pdn_conn_dyn_params *ret_dyn_info)
+{
+	struct at_param_list param_list = { 0 };
+	size_t param_str_len;
+	char *next_param_str;
+	char at_cmd_str[16];
+	char at_cmd_response_str[1024];
+	char dns_addr_str[256];
+	char *at_ptr = at_cmd_response_str;
+	int ret, family, iterator;
+	struct in_addr *addr;
+	struct in6_addr *addr6;
+	bool resp_continues = true;
+
+	sprintf(at_cmd_str, AT_CMD_PDP_CONTEXT_READ_DYN_INFO, cid);
+
+	ret = nrf_modem_at_cmd(at_cmd_response_str, sizeof(at_cmd_response_str), at_cmd_str);
+	if (ret) {
+		mosh_error("Cannot get PDP conn dyn params, ret: %d", ret);
+		return false;
+	}
+
+	/* Parse the response */
+	ret = at_params_list_init(&param_list, AT_CMD_PDP_CONTEXT_READ_DYN_INFO_PARAM_COUNT);
+	if (ret) {
+		mosh_error("Could not init AT params list, error: %d\n", ret);
+		return ret;
+	}
+
+	iterator = 0;
+	while (resp_continues) {
+		resp_continues = false;
+		ret = at_parser_max_params_from_str(at_ptr, &next_param_str, &param_list,
+						    AT_CMD_PDP_CONTEXT_READ_DYN_INFO_PARAM_COUNT);
+		if (ret == -EAGAIN) {
+			resp_continues = true;
+		} else if (ret == -E2BIG) {
+			mosh_error("E2BIG, error: %d\n", ret);
+		} else if (ret != 0) {
+			mosh_error("Could not parse AT response for %s, error: %d\n", at_cmd_str,
+				   ret);
+			goto clean_exit;
+		}
+
+		/* Read primary DNS address */
+		param_str_len = sizeof(dns_addr_str);
+		ret = at_params_string_get(&param_list,
+					   AT_CMD_PDP_CONTEXT_READ_DYN_INFO_DNS_ADDR_PRIMARY_INDEX,
+					   dns_addr_str, &param_str_len);
+		if (ret) {
+			mosh_error("Could not parse dns str for cid %d, err: %d", cid, ret);
+			goto clean_exit;
+		}
+		dns_addr_str[param_str_len] = '\0';
+
+		if (dns_addr_str != NULL) {
```

Comment:
You are checking if dns_addr_str is not NULL here, but you already manipulate its contents above. I don't know if it actually can ever be NULL, but maybe you could at least move dns_addr_str[param_str_len] = '\0'; inside this if?

Reply:
Thanks. Removed as unnecessary.